app/controllers/assets_controller.rb

```diff
@@ -48,17 +48,24 @@ def file_preview
       'type' => (@asset.is_image? ? 'image' : 'file'),
 
       'filename' => truncate(@asset.file_file_name,
-                             length:
-                               Constants::FILENAME_TRUNCATION_LENGTH),
-      'download-url' => download_asset_path(@asset, timestamp: Time.now.to_i),
-      'editable'     => @asset.editable?(current_user)
+                             length: Constants::FILENAME_TRUNCATION_LENGTH),
+      'download-url' => download_asset_path(@asset, timestamp: Time.now.to_i)
     }
 
+    can_edit = if @assoc.class == Step
+                 can_manage_protocol_in_module?(@protocol) || can_manage_protocol_in_repository?(@protocol)
+               elsif @assoc.class == Result
+                 can_manage_module?(@my_module)
+               elsif @assoc.class == RepositoryCell
+                 can_manage_repository_rows?(@repository.team)
+               end
```

Comment:
As a piece of mind, would it be better if we had additional 'else false` here? Just thinking about if this could be abused by anyone.

Reply:
As far as I can remember nil equals to false in 'if' checks, so if can_edit is not set to true, it will fail in latter check